src/test/coinsviewcacheasync_tests.cpp

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <coins.h>
+#include <coinsviewcacheasync.h>
+#include <primitives/block.h>
+#include <primitives/transaction.h>
+#include <primitives/transaction_identifier.h>
+#include <test/util/random.h>
+#include <test/util/setup_common.h>
+#include <uint256.h>
+#include <util/hasher.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <cstdint>
+#include <memory>
+#include <ranges>
+#include <unordered_set>
+
+BOOST_AUTO_TEST_SUITE(coinsviewcacheasync_tests)
+
+struct NoAccessCoinsView : CCoinsView {
+    std::optional<Coin> GetCoin(const COutPoint&) const override { abort(); }
+};
+
+struct CoinsViewCacheAsyncTest : BasicTestingSetup {
+private:
+    std::unique_ptr<CoinsViewCacheAsync> m_async_cache{nullptr};
+    std::unique_ptr<CBlock> m_block{nullptr};
+
+    CBlock CreateBlock(int32_t num_txs) const noexcept
```

Comment:
we're only using `100` here, we might as well:

```suggestion
    static constexpr auto num_txs{100};
    CBlock CreateBlock() const noexcept
```